src/common/dav_v2/tx.c

```diff
@@ -1854,9 +1854,13 @@ dav_allot_mb_evictable_v2(dav_obj_t *pop, int flags)
 	D_ASSERT(flags == 0);
 	D_ASSERT((dav_tx_stage_v2() == DAV_TX_STAGE_NONE));
 
+	err = lw_tx_begin(pop);
```

Comment:
btw, lw_tx_begin() could yield too (when WAL is running out of space, it needs to wait space reclaiming by checkpointing).

Reply:
Yes, this change will take care of this. The the check for active_evictable_mb being set in heap_get_evictable_mb() will prevent creating evictable mbs concurrently.